tests/security_tests.rs

```diff
@@ -0,0 +1,195 @@
+/// |  Block Integrity Block                 |   11  |    2   |
+/// |  OP(bib-integrity, target=1)           |       |        |
+/// +----------------------------------------+-------+--------+
+/// |  Payload Block                         |   1   |    1   |
+/// +----------------------------------------+-------+--------+
+/// ```
+///
+/// see rfc for more details:
+/// https://www.rfc-editor.org/rfc/rfc9173.html#name-example-1-simple-integrity
+fn simple_integrity_test(){
+
+    println!("Simple Integrity Test");
+
+
+
+    // Create Original bundle
+    let dst = eid::EndpointID::with_ipn(1,2).unwrap();
+    let src = eid::EndpointID::with_ipn(2,1).unwrap();
+    let now = dtntime::CreationTimestamp::with_time_and_seq(0, 40);
+    
+    let primary_block = primary::PrimaryBlockBuilder::default()
+        .destination(dst)
+        .source(src.clone())
+        .report_to(src)
+        .creation_timestamp(now)
+        .lifetime(Duration::from_millis(1000000))
+        .build()
+        .unwrap();
+    let cbor_primary = serde_cbor::to_vec(&primary_block).unwrap();
+    let cbor_primary = hexify(&cbor_primary);
+    let example_cbor_primary = "88070000820282010282028202018202820201820018281a000f4240";
+    assert_eq!(cbor_primary, example_cbor_primary);
+
+    let payload_block = bp7::new_payload_block(
+        BlockControlFlags::empty(), 
+        b"Ready to generate a 32-byte payload".to_vec());
+    let cbor_payload = serde_cbor::to_vec(&payload_block).unwrap();
+    let cbor_payload = hexify(&cbor_payload);
+    let example_cbor_payload = "85010100005823526561647920746f2067656e657261746520612033322d62797465207061796c6f6164";
+    assert_eq!(cbor_payload, example_cbor_payload);
+
+    // Create Block Integrity Block
+    // Two Parts: First create IPPT then ASB
+    
+    // First Create Integrity-Protected Plaintext
+    let sec_block_header: (CanonicalBlockType, u64, bp7::flags::BlockControlFlagsType) = (bp7::security::INTEGRITY_BLOCK, 2, BlockControlFlags::empty().bits());
+    
+    let sec_ctx_para = BibSecurityContextParameter {
+        sha_variant: Some((1, HMAC_SHA_512)), 
+        wrapped_key: None,
+        integrity_scope_flags: Some((3, 0x0000)),
+    };
+
+    let mut ippt = bp7::security::IpptBuilder::default()
+        .primary_block(primary_block.clone())  
+        .security_header(sec_block_header)
+        .scope_flags(0x0000)
+        .build();
+    let ippt_complete = ippt.create(&payload_block);
+    let ippt_list = vec![(payload_block.block_number, &ippt_complete)];
+
+    let cbor_ippt_complete = hexify(&ippt_complete);
+    let example_ippt_complete = "005823526561647920746f2067656e657261746520612033322d62797465207061796c6f6164";
+    //println!("{:?}", cbor_ippt_complete);
+    
+    assert_eq!(cbor_ippt_complete, example_ippt_complete);
+
+
+    // Second Create Abstract Security Block
+    let sec_ctx_para = bp7::security::BibSecurityContextParameter::new(Some((1,7)),None,Some((3,0x0000)));
```

Comment:
maybe the magic numbers from the test should also be explained directly in a comment like 1,7,3 etc